File: Server/Subscriptions/Entities/Subscription.cs
using Subscriptions.Enum;

namespace Subscriptions.Entities
{
    public class Subscription
    {
        public Guid Id { get; set; }

        public Guid UserId { get; set; }  

        public string Name { get; set; } = string.Empty;

        public decimal Amount { get; set; }

        public string Category { get; set; } = "Other";

        public BillingStatus Status { get; set; } = BillingStatus.Active;

        public BillingCycle BillingCycle { get; set; }

        public DateTime NextBillingDate { get; set; } = DateTime.MinValue;

    }
}

File: Client/src/Types/subscription.ts
export interface subscription{
    Name : string;
    Amount: number;
    BillingCycle: string;
    NextBillingDate: string;
    Category: string;
}

export interface SubscriptionDto {
  id: string;
  name: string;
  amount: number;
  billingCycle: string;
  status: string;
  nextBillingDate: string;
  category: string;
}